src/main.rs

```diff
@@ -238,3 +256,304 @@ fn parse_args() -> Result<LndCfg, ArgsError> {
+        peers_client_mock
+            .expect_update_node_announcement()
+            .returning(|_| {
+                Err(SetOnionBitError::UpdateAnnouncementErr(Status::new(
+                    Code::Unimplemented,
+                    "",
+                )))
+            });
+
+        let set_feature_err = set_feature_bit(&mut client_mock, &mut peers_client_mock)
+            .await
+            .expect_err("set_feature_bit should error with unavailable");
+
+        // If the peers client returns Code::Unimplemented, we should get
+        // the correct error message.
+        matches!(set_feature_err, SetOnionBitError::UnimplementedPeersService);
+    }
+
+    #[tokio::test]
+    async fn test_check_if_onion_message_set_failure() {
+        let mut client_mock = MockInfoRetriever::new();
+        let mut peers_client_mock = MockNodeAnnouncementUpdater::new();
+
+        peers_client_mock
+            .expect_update_node_announcement()
+            .returning(|_| Ok(()));
+
+        client_mock.expect_get_info().returning(|_| {
+            Err(SetOnionBitError::UpdateAnnouncementErr(Status::new(
+                Code::Unavailable,
+                "",
+            )))
+        });
+
+        let set_feature_err = set_feature_bit(&mut client_mock, &mut peers_client_mock)
+            .await
+            .expect_err("set_feature_bit should error with unavailable");
+
+        matches!(set_feature_err, SetOnionBitError::UpdateAnnouncementErr(_));
+
+        // If get_info returns a response with the wrong feature bit set,
+        // set_feature_bit should throw another error.
+        client_mock.expect_get_info().returning(|_| {
+            Ok(tonic_lnd::lnrpc::GetInfoResponse {
+                features: HashMap::from([(
+                    8,
+                    tonic_lnd::lnrpc::Feature {
+                        name: String::from("testing"),
+                        is_known: true,
+                        is_required: false,
+                    },
+                )]),
+                alias: Default::default(),
+                best_header_timestamp: Default::default(),
+                block_hash: Default::default(),
+                block_height: Default::default(),
+                chains: Default::default(),
+                color: Default::default(),
+                commit_hash: Default::default(),
+                identity_pubkey: Default::default(),
+                num_active_channels: Default::default(),
+                num_inactive_channels: Default::default(),
+                num_peers: Default::default(),
+                num_pending_channels: Default::default(),
+                require_htlc_interceptor: Default::default(),
+                synced_to_chain: Default::default(),
+                synced_to_graph: Default::default(),
+                testnet: Default::default(),
+                uris: Default::default(),
+                version: Default::default(),
```

Comment:
nit: so a nice thing we can do here to reduce the boilerplate and also avoid any deprecation warnings is to use the [struct update syntax](https://doc.rust-lang.org/stable/book/ch05-01-defining-structs.html#creating-instances-from-other-instances-with-struct-update-syntax) with `Default::default()`
```suggestion
                features: HashMap::from([(
                    8,
                    tonic_lnd::lnrpc::Feature {
                        name: String::from("testing"),
                        is_known: true,
                        is_required: false,
                    },
                )]),
                ..Default::default()
```

Reply:
Ooohh this is awesome! Thanks for the tip

Reply:
Updated, looks so much cleaner B)